apps/els_dap/src/els_dap_general_provider.erl

```diff
@@ -1084,3 +1091,25 @@ distribution_error(Error) ->
             io_lib:format("Could not start Erlang distribution. ~p", [Error])
         )
     ).
+
+-spec maybe_interprete_and_clear_module(node(), module()) -> {boolean(), binary()}.
+maybe_interprete_and_clear_module(ProjectNode, Module) ->
+    case els_dap_rpc:interpretable(ProjectNode, Module) of
+        true ->
+            {module, Module} = els_dap_rpc:i(ProjectNode, Module),
+
+            %% purge all breakpoints from the module
```

Comment:
Why do we want to purge them?

Reply:
It's just easier to code this way. The DAP protocol gives the following information on changes to breakpoints:
- complete set of line breakpoints per module
- complete set of function breakpoints in total

So we delete all breakpoints and set them again.

We should probably calculate a delta and then use `delete_break` and `del_break_in`. This PR doesn't change the general way we set breakpoints, just makes sure we don't crash on invalid ones.